llvm/lib/Target/X86/X86AsmPrinter.cpp

```diff
@@ -918,8 +918,11 @@ void X86AsmPrinter::emitStartOfAsmFile(Module &M) {
   // If this is not inline asm and we're in 16-bit
   // mode prefix assembly with .code16.
   bool is16 = TT.getEnvironment() == Triple::CODE16;
-  if (M.getModuleInlineAsm().empty() && is16)
-    OutStreamer->emitAssemblerFlag(MCAF_Code16);
+  if (M.getModuleInlineAsm().empty() && is16) {
+    auto *XTS =
+        static_cast<X86TargetStreamer *>(OutStreamer->getTargetStreamer());
```

Comment:
X86AsmPrinter should probably define a convenience method that returns X86TargetStreamer

Reply:
Yeah, these targets probably should. I encourage those who care about these backends to do cleanup work there :) Similarly, it would be nice if ARMELFStreamer didn't have its own IsThumb state to keep in sync with the actual subtarget. But those kinds of cleanups are future work.